kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs

```diff
@@ -189,11 +189,35 @@ impl CodegenBackend for GotocCodegenBackend {
 
     fn link(
         &self,
-        _sess: &Session,
-        _codegen_results: CodegenResults,
-        _outputs: &OutputFilenames,
+        sess: &Session,
+        codegen_results: CodegenResults,
+        outputs: &OutputFilenames,
     ) -> Result<(), ErrorGuaranteed> {
-        Ok(())
+        // In `link`, we need to do two things:
+        // 1. We need to emit `rlib` files normally. Cargo expects these in some circumstances and sends
+        //    them to subsequent builds with `-L`.
+        // 2. We MUST NOT try to invoke the native linker, because that will fail. We don't have real objects.
+        // This is normally not a problem: usually we only get one requested `crate-type`.
+        // But let's be careful and fail loudly if we get conflicting requests:
+        let requested_crate_types = sess.crate_types();
+        // Quit successfully if we don't need an Rlib:
```

Comment:
```suggestion
        // Quit successfully if we don't need an `rlib`:
```
(as in other mentions)